src/load

```diff
@@ -398,25 +404,24 @@ def save_epss_data(appctx, args, db_records):
         bar_format = '{n_fmt}/{total} {l_bar}{bar}| ({elapsed}/{remaining})'
         db_insert_progress = tqdm(total=len(db_records), bar_format=f'{db_insert_progress_name:<20} {bar_format}',ascii=True)
 
-        for batch in [list(db_records.keys())[i:i + BATCH_SIZE] for i in range(0, len(db_records), BATCH_SIZE)]:
+        BATCH_SIZE_epss = 1024
+        for batch in [list(db_records.keys())[i:i + BATCH_SIZE_epss] for i in range(0, len(db_records), BATCH_SIZE_epss)]:
 
-            vuln_query= session.query(Vuln).filter(Vuln.vuln_id.in_(batch)).options(load_only(Vuln.data))
+            vuln_query= session.query(Vuln).filter(Vuln.vuln_id.in_(batch)).options(load_only(Vuln.data)).all()
             update_values = []
 
-            for vuln_record in vuln_query.all():
-
+            for vuln_record in vuln_query:
                 epss_desired_row = db_records[vuln_record.vuln_id]       # Find the corresponding record in the batch
 
                 if 'epss' in vuln_record.data['metrics']:           # Check if 'epss' key already exists in data.metrics
                     vuln_record.data['metrics']['epss'].update({'score': epss_desired_row[0],'percentile': epss_desired_row[1],'date': epss_desired_row[2]})
                 else:
                     vuln_record.data['metrics']['epss']= {'score': epss_desired_row[0],'percentile': epss_desired_row[1],'date': epss_desired_row[2]}
 
-                update_dict = {'id': vuln_record.id, 'data': vuln_record.data}
-                update_values.append(update_dict)
-            session.bulk_update_mappings(Vuln, update_values)
+                update_values.append({'id': vuln_record.id, 'data': vuln_record.data})
 
-            db_insert_progress.update(len(batch))
+            session.bulk_update_mappings(Vuln, update_values)
+            db_insert_progress.update(BATCH_SIZE_epss)
```

Comment:
The update of the progress should be based on the batch length that is beging processed and not on the batch size that the original array is split on. (The last batch will most probably have a shorter length than BATCH SIZE which will lead incorrect progress show at the end.